### PR TITLE
[BD-18] Standardize the make test targets

### PR DESCRIPTION
**Pull request by emzubair**
## Standardize the make test targets
Added target for test & clean

**Packages added**

- pytest-cov

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @emzubair! I've created [BLENDED-460](https://openedx.atlassian.net/browse/BLENDED-460) to keep track of it in Jira.  More details are on the [BD-18](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1593934815/BD-18+Testing+Consistency+Improvements) project page.

When this pull request is ready, tag your edX technical lead.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,15 @@
+.PHONY: clean test upgrade test
```
Comment (feanil) [inline Makefile:1]: Add the rest of the targets to `.PHONY`

File: Makefile [hunk 2]
```diff
@@ -9,4 +21,19 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
\ No newline at end of file
+	mv requirements/test.tmp requirements/test.txt
+
+requirements: ## install development environment requirements
+	pip install -q -r requirements/pip_tools.txt
+	pip install -qr requirements/base.txt --exists-action w
+	pip-sync requirements/base.txt requirements/constraints.txt requirements/test.txt requirements/travis.txt
+
+test-python: clean ## run tests using pytest and generate coverage report
+	pytest
```
Comment (feanil) [inline Makefile:32]: Use tox to run tests instead.

File: Makefile [hunk 3]
```diff
@@ -9,4 +21,19 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
+
+quality-python: ## Run python linters
```
Comment (feanil) [inline Makefile:34]: Add edx-lint to this repo for linting.

**Comment by emzubair** [comment]
@feanil updated the `PR`

**Pull request merged**